beets/plugins.py

```diff
@@ -202,9 +244,9 @@ def add_media_field(self, name, descriptor):
         library.Item._media_fields.add(name)
 
     _raw_listeners = None
-    listeners = None
+    listeners: None | dict[str, list[Callable]] = None
```

Comment:
Maybe add a `Protocol` and use it here and below:
```py
class Listener(typing.Protocol):
    def __call__(self, **kwargs) -> Any: ...

class BeetsPlugin:
    # ...
    listeners = None | dict[str, Listener]
```

(Looking at `send()`, it appears that the only restriction that is generally true for event listeners is that they only receive arguments by keyword.)

Additionally, `listeners` and `_raw_listeners` should have the same type here (since the only difference is wrapping the function in a way that should not alter the signature).

Reply:
Yeah we could definitely do this, I was just too lazy tbh. I think a typed callable should be fine too, maybe a bit less code than using the full protocol definition.

Will have a look later today.